### PR TITLE
log connection failures

### PR DESCRIPTION
**Pull request by mark-prins**
Helps with #4383 
Fixes #Nothing sorry

## Change summary

Just added logging to the areas which throw the annoying 'unable to connect'
Had to typescript the logger a little, since one of the methods was already a ts file.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Initialise a tablet with an invalid URL and check that a log is created

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

File: src/utilities/logging/Engine.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+//     debug: (text: string) => void
+//     info: (text: string) => void
+//     warn: (text: string) => void
+//     error: (text: string) => void
+//     fatal: (text: string) => void
+//     setLogLevel(transportKey: string, newLevel: int) => void
+// }
+
+export class BunyanLoggingEngine {
+  bunyan: null | any = null;
```
Comment (mark-prins) [inline src/utilities/logging/Engine.ts:19]: didn't try very hard to correctly type the bunyan as it is only referenced in the `LoggerService`

File: src/utilities/logging/index.ts [hunk 2]
```diff
@@ -30,7 +32,7 @@ export const tidyLogFiles = async () => {
   } catch (error) {
     const storeCode = UIDatabase.getSetting(SETTINGS_KEYS.THIS_STORE_CODE);
     const syncUrl = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_URL);
-    Bugsnag.notify(error, content => {
+    Bugsnag.notify(error as NotifiableError, (content: any) => {
```
Comment (mark-prins) [inline src/utilities/logging/index.ts:35]: should have tried harder to type the `content` parameter. Thought it didn't matter much 🤷

**Review by katherine-sussol — APPROVED** [review]
👍  🌳 🪓  

I've only tested locally and haven't been able to see much except 'Network request failed' when I turn off the server or use incorrect URLs, but presumably, there may be some more useful things logged in other circumstances? Think the visibility is still good 🤷‍♀️

File: src/utilities/logging/Engine.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  trace: Action = (text, details) => {
+    this.bunyan?.trace(text, details);
+  };
+
+  debug: Action = (text, details) => {
+    this.bunyan?.debug(text, details);
+  };
+
+  info: Action = (text, details) => {
+    this.bunyan.info(text), details;
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply src/utilities/logging/Engine.ts:42]: thanks!

**Pull request merged**